Source/Core/Core/PowerPC/Jit64/Jit_Integer.cpp

```diff
@@ -1272,54 +1275,58 @@ void Jit64::divwx(UGeckoInstruction inst)
   JITDISABLE(bJITIntegerOff);
   int a = inst.RA, b = inst.RB, d = inst.RD;
 
-  if (gpr.R(a).IsImm() && gpr.R(b).IsImm())
+  auto ra = regs.gpr.Lock(a);
+  auto rb = regs.gpr.Lock(b);
+  auto rd = regs.gpr.Lock(d);
+
+  if (ra.IsImm() && rb.IsImm())
   {
-    s32 i = gpr.R(a).SImm32(), j = gpr.R(b).SImm32();
+    s32 i = ra.SImm32(), j = rb.SImm32();
     if (j == 0 || (i == (s32)0x80000000 && j == -1))
     {
-      gpr.SetImmediate32(d, (i >> 31) ^ j);
+      rd.SetImm32((i >> 31) ^ j);
       if (inst.OE)
         GenerateConstantOverflow(true);
     }
     else
     {
-      gpr.SetImmediate32(d, i / j);
+      rd.SetImm32(i / j);
       if (inst.OE)
         GenerateConstantOverflow(false);
     }
   }
   else
   {
-    gpr.Lock(a, b, d);
     // no register choice
-    gpr.FlushLockX(EAX, EDX);
-    gpr.BindToRegister(d, (d == a || d == b), true);
-    MOV(32, R(EAX), gpr.R(a));
+    auto eax = regs.gpr.Borrow(EAX);
+    auto edx = regs.gpr.Borrow(EDX);
```

Comment:
I guess this works out because EAX and EDX happen to be the scratch registers?  You might want to note that explicitly.